torchvision/models/detection/backbone_utils.py

```diff
@@ -65,16 +62,24 @@ def resnet_fpn_backbone(backbone_name, pretrained, norm_layer=misc_nn_ops.Frozen
         >>>    ('pool', torch.Size([1, 256, 1, 1]))]
 
     Arguments:
-        backbone_name (string): resnet architecture. Possible values are 'ResNet', 'resnet18', 'resnet34', 'resnet50',
-             'resnet101', 'resnet152', 'resnext50_32x4d', 'resnext101_32x8d', 'wide_resnet50_2', 'wide_resnet101_2'
+        backbone (string or callable): resnet architecture. Possible values are 'ResNet',
+            'resnet18', 'resnet34', 'resnet50', 'resnet101', 'resnet152', 'resnext50_32x4d',
+            'resnext101_32x8d', 'wide_resnet50_2', 'wide_resnet101_2'; you can also pass in your own
+            callable, e.g. functools.partial(torch.hub.load, 'zhanghang1989/ResNeSt', 'resnest50').
         norm_layer (torchvision.ops): it is recommended to use the default value. For details visit:
-            (https://github.com/facebookresearch/maskrcnn-benchmark/issues/267)
-        pretrained (bool): If True, returns a model with backbone pre-trained on Imagenet
+            (https://github.com/facebookresearch/maskrcnn-benchmark/issues/267).
+        pretrained (bool): If True, returns a model with backbone pre-trained on Imagenet.
         trainable_layers (int): number of trainable (not frozen) resnet layers starting from final block.
             Valid values are between 0 and 5, with 5 meaning all backbone layers are trainable.
     """
+    if isinstance(backbone, str):
+        backbone = resnet.__dict__[backbone](pretrained=pretrained, norm_layer=norm_layer)
+    elif callable(backbone):
+        backbone = backbone(pretrained=pretrained, norm_layer=norm_layer)
```

Comment:
This assumes that the callable has a particular interface and its modules follow the same convention of resnets, which is not always the case and will fail for most of the models in torchvision. As such, I think it might be better to avoid trying to be generic in here, given that this function is not meant to be generic.